Makefile

```diff
@@ -734,6 +734,7 @@ TEST_BUILTINS_OBJS += test-serve-v2.o
 TEST_BUILTINS_OBJS += test-sha1.o
 TEST_BUILTINS_OBJS += test-sha256.o
 TEST_BUILTINS_OBJS += test-sigchain.o
+TEST_BUILTINS_OBJS += test-simple-ipc.o
```

Comment:
It would be nice to have this helper a lot earlier, but I guess it can come at the end, too.

Reply:
The test isn't usable until at least the client side implementation is present (and even then it would only be a trivial test).  So I'm not sure how much I could mover earlier.

Would it help to move it after the first implementation (windows) and before the unix implementation ??

Reply:
I moved the test to be right after the windows implementation and before the unix one.

Reply:
I think that's fair.

As to the test being usable only when at least the client side is implemented: yes, and the test has to be skipped if the client side is not available (which, for the time being, is the case on all BSDs, IRIXes, HPUXes, etc). 😀

Reply:
I had to move the test to the last commit to get intermediate commits to build on mac/linux.  i think i should guard the test declaration like we did the compat/simple-ipc/*.c files and in the makefile.

Reply:
I found a way to simplify this and have the windows commit, the test commit, and then the unix commit so that intermediate commits don't fail to build in unix -- but i had to squash the client and server commits together.